docs/user/agent/node-joiner.sh

```diff
@@ -0,0 +1,131 @@
+#!/bin/bash
+
+if [ $# -lt 1 ]; then
+    echo "./node-joiner.sh <pull secret path>"
+    echo "Usage example:"
+    echo "$ ./node-joiner.sh ~/config/my-pull-secret"
+
+    exit 1
+fi
+pullSecret=$1
+
+# Extract the installer image pullspec and release version.
+releaseImage=$(oc get clusterversion version -o=jsonpath='{.status.history[?(@.state == "Completed")].image}')
+nodeJoinerPullspec=$(oc adm release info -a "$pullSecret" --image-for=installer "$releaseImage")
+
+# Create the namespace to run the node-joiner, along with the required roles and bindings.
+staticResources=$(cat <<EOF
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: openshift-node-joiner
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: node-joiner
+  namespace: openshift-node-joiner
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: node-joiner
+rules:
+- apiGroups:
+  - config.openshift.io
+  resources:
+  - clusterversions
+  - proxies
+  verbs:
+  - get
+- apiGroups:
+  - ""
+  resources:
+  - secrets
+  - configmaps
+  - nodes
+  verbs:
+  - get
+  - list
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: node-joiner
+subjects:
+- kind: ServiceAccount
+  name: node-joiner
+  namespace: openshift-node-joiner
+roleRef:
+  kind: ClusterRole
+  name: node-joiner
+  apiGroup: rbac.authorization.k8s.io
+EOF
+)
+echo "$staticResources" | oc apply -f -
+
+# Generate a configMap to store the user configuration
+oc create configmap nodes-config --from-file=nodes-config.yaml -n openshift-node-joiner -o yaml --dry-run=client | oc apply -f -
+
+# Runt the node-joiner pod to generate the ISO
+nodeJoinerPod=$(cat <<EOF
+apiVersion: v1
+kind: Pod
+metadata:
+  name: node-joiner
+  namespace: openshift-node-joiner
+  annotations:
+    openshift.io/scc: anyuid
+  labels:
+    app: node-joiner    
+spec:
+  restartPolicy: Never
+  serviceAccountName: node-joiner
+  securityContext:
+    seccompProfile:
+      type: RuntimeDefault
+  containers:
+  - name: node-joiner
+    imagePullPolicy: IfNotPresent
+    image: $nodeJoinerPullspec
+    volumeMounts:
+    - name: nodes-config
+      mountPath: /config
+    - name: assets
+      mountPath: /assets
+    command: ["/bin/sh", "-c", "cp /config/nodes-config.yaml /assets; HOME=/assets node-joiner add-nodes --dir=/assets --log-level=debug; echo \$? > /assets/completed; sleep 600"]    
```

Comment:
It looks like there's an opportunity here to modify the command to allow separate input and output directories, and have some sort of built-in signalling/waiting mechanism.